site/src/pages/CreateWorkspacePage/CreateWorkspacePageView.tsx

```diff
@@ -223,64 +218,40 @@ export const CreateWorkspacePageView: FC<CreateWorkspacePageViewProps> = ({
           </FormSection>
 
           {parameters && (
-            <>
-              <MutableTemplateParametersSection
-                templateParameters={parameters}
-                getInputProps={(parameter, index) => {
-                  return {
-                    ...getFieldHelpers(
-                      "rich_parameter_values[" + index + "].value",
-                    ),
-                    onChange: async (value) => {
-                      await form.setFieldValue(
-                        "rich_parameter_values." + index,
-                        {
-                          name: parameter.name,
-                          value: value,
-                        },
-                      );
-                    },
-                    disabled:
-                      disabledParamsList?.includes(
-                        parameter.name.toLowerCase().replace(/ /g, "_"),
-                      ) || creatingWorkspace,
-                  };
-                }}
-              />
-              <ImmutableTemplateParametersSection
-                templateParameters={parameters}
-                classes={{
-                  root: css`
-                    border: 1px solid ${theme.palette.warning.light};
-                    border-radius: 8px;
-                    background-color: ${theme.palette.background.paper};
-                    padding: 80px;
-                    margin-left: -80px;
-                    margin-right: -80px;
-                  `,
-                }}
-                getInputProps={(parameter, index) => {
-                  return {
-                    ...getFieldHelpers(
-                      "rich_parameter_values[" + index + "].value",
-                    ),
-                    onChange: async (value) => {
-                      await form.setFieldValue(
-                        "rich_parameter_values." + index,
-                        {
+            <FormSection
+              title="Parameters"
+              description="These are the settings used by your template. Please note that immutable parameters cannot be modified once the workspace is created."
+            >
+              {/*
+                Opted not to use FormFields in order to increase spacing.
+                This decision was made because rich parameter inputs are more visually dense than standard text fields.
+              */}
+              <div css={{ display: "flex", flexDirection: "column", gap: 36 }}>
+                {parameters.map((parameter, index) => {
```

Comment:
NIT: I might lift this function outside of the JSX.

Reply:
Which function? The map?

Reply:
the callback that the map takes

Reply:
I think it is quite normal in react codebases but any specific concern?

Reply:
I feel like it lowers general readability because its a longer callback that defines multiple variables, but no strong concerns!

Reply:
I see! But in this case, we would have to pass 3+ arguments to the function so I'm not sure if extracting it to a new function or component would improve that much but I can understand the reasoning.

Reply:
good point, good point